main.rb

```diff
@@ -0,0 +1,120 @@
+require './person'
+require './student'
+require './teacher'
+require './book'
+require './rental'
+
+def list_books(books)
+  books.each do |book|
+    puts "Title: \"#{book.title}\", Author: #{book.author}"
+  end
+end
+
+def list_people(people)
+  people.each do |person|
+    puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+end
+
+def create_book(books)
+  puts 'Title: '
+  title = gets.chomp
+  puts 'Author: '
+  author = gets.chomp
+  book = Book.new(title, author)
+  books.push(book)
+  puts 'Book created successfully!' if books.include?(book)
+end
+
+def create_person(people)
+  print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+  option = gets.chomp
+  puts 'Age: '
+  age = gets.chomp.to_i
+  puts 'Name: '
+  name = gets.chomp
+  person = nil
+  case option
+  when '1'
+    puts 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    permission = (permission.downcase == 'y')
+    person = Student.new(age, name, permission)
+  when '2'
+    puts 'Specialization:'
+    specialization = gets.chomp
+    person = Teacher.new(age, specialization, name)
+  end
+  people.push(person)
+  puts 'Person created successfully!' if people.include?(person)
+end
```

Comment:
- [ ] Creating a person object is failing. I think it's better if we try to fix this. I recommend you use only keyword arguments if that helps `keyword:`. An example of creating a student would be `Student.new(age: age, name: name, permission: permission)`. 

If you want to keep the classic no-named arguments, make sure that if you any keyword argument on the class definitions that you pass it correctly when you instantiate the class.

![school-failing-student](https://user-images.githubusercontent.com/79114233/146469842-5518139f-ff77-40d0-a7f4-63a48f9a6c34.gif)

Reply:
Note to the next reviewer: Make sure to test for creating rentals and listing them, as I can't crate rentals without creating persons.